src/graph.ts

```diff
@@ -849,54 +849,44 @@ export class Graph {
                 });
             });
 
-            var visitedEdges:Set<string> = new Set();
             var visitedEdgeList:string[] = [];
             
             if(matches['matchings'] &&  matches['matchings'].length > 0) {
                 
                 var match = matches['matchings'][0];
 
-                // this is kind of convoluted due to the sparse info returned in the OSRM annotations...
-                // write out sequence of nodes and edges as emitted from walking OSRM-returned nodes
-                // finding the actual posistion and directionality of the OSRM-edge within the ShSt graph 
-                // edge means that we have to snap start/end points in the OSRM geom
+                // This is a bit messy - we need to go through the legs of the matched journey
+                // and find the edges that we passed through in order
+                // for each leg, iterate through the pairs of nodes and find the corresponding edge
+                // if it's the same as the previous edge, then ignore it
```

Comment:
> // if it's the same as the previous edge, then ignore it

just curious, did these scenarios really happen in your tests, i.e. OSRM returning the same edge twice?

Reply:
It does happen - but just because there are several OSRM nodes that exist within the same shared streets edge. Maybe a better way of saying this would be "if we are still on the same shst ref, don't bother adding it again".